fbpic/particles/numba_methods.py

```diff
@@ -133,29 +133,24 @@ def gather_field_numba(exptheta, m, Fgrid, Fptcl,
     for ip in range(Ntot):
         # Erase the temporary variable
         F = 0.j
-        # Create help variables for the right index and shape. This is necessary
-        # because otherwise the signum in the z-component will be wrong
-        # because this method gets called three times. If the indices were
-        # to be corrected in the first call of this function by doing
-        # ir[cell_index_r][ip] = abs(ir[cell_index_r][ip]) - 1
-        # then for the second and thrid call the signum would never change
-        # but it has to in the thrid call (since the z-component has a
-        # different signum change)
-        ir_corr = 0
-        Sr_corr = 0
         # Loop over all the adjacent cells (given by shape order)
+        # Use helper variables `ir_corr` and `Sr_corr`.
+        # This is necessary, because ir and Sr should **not** be modified
+        # **in-place**. (This is because ir and Sr are reused several
+        # times, as we call the present function 3 times, with different
+        # values for sign_guards.)
         for cell_index_r in range(ir.shape[0]):
             for cell_index_z in range(iz.shape[0]):
                 # Correct the guard cell index and sign
-                if ir[cell_index_r][ip] < 0:
-                    ir_corr = abs(ir[cell_index_r][ip]) - 1
-                    Sr_corr = sign_guards * Sr[cell_index_r][ip]
+                if ir[cell_index_r, ip] < 0:
+                    ir_corr = abs(ir[cell_index_r, ip]) - 1
+                    Sr_corr = sign_guards * Sr[cell_index_r, ip]
                 else:
-                    ir_corr = ir[cell_index_r][ip]
-                    Sr_corr = Sr[cell_index_r][ip]
+                    ir_corr = ir[cell_index_r, ip]
+                    Sr_corr = Sr[cell_index_r, ip]
                 # Gather the field value at the respective grid point
-                F += Sz[cell_index_z][ip] * Sr_corr * \
-                    Fgrid[iz[cell_index_z][ip], ir_corr]
+                F += Sz[cell_index_z, ip] * Sr_corr * \
+                    Fgrid[ iz[cell_index_z, ip], ir_corr]
```

Comment:
The above changes are just esthetical, and are here for consistency with the rest of the code: we usually use `,` instead of `][` when accessing 2darrays. Sorry again for being annoyingly picky!!